### PR TITLE
feat: nav menu

### LINKED ISSUE #65: Routes file structure
**Changes needed**
Setup tentative route file structure.

### PR DESCRIPTION
**Pull request by AasmundN**
This PR is meant to give us a nice starting point for further development. It does this by adding some essential "infrastructure" to the application, such as a navigation menu and tentative route setup :love_you_gesture: :pray:

# Changes
- Add horizontal and vertical nav menu, for wide and narrow screens respectively.
- Remove `@protected` folder. The meta data must now be added manually instead. This proved to be a cleaner approach.
- Setup tentative route files. Resolves #65.
- Add page fade transition :wind_face:
- There is a weird bug in _Vuetify_ when using SSR that makes the _VMain_ content render under the _VAppBar_. The content would then transition into place. A workaround was added; a static padding matching the height of the app bar.

### CODE REVIEW

**Comment by ipeglin** [comment]
Looks like there is some trouble with the navigation in mobile view. For some reason you have to double click the hamburger menu to toggle it after rerouting to a page. 

** Expected behaviour **
Clicking on the hamburger icon toggles the menu, and it slides in from the right. When navigating to a page this goes as expected by clicking on any of the links. When you then want to navigate to a yet another route. Click on the hamburger once again, and then on the route link.

** Recreation **
1. Load dev project
2. Scale to mobile view
3. Toggle hamburger menu
4. Navigate to any route
5. Try to navigate to another route. You should now notice you need to click the hamburger twice.

**Comment by AasmundN** [comment]
> Looks like there is some trouble with the navigation in mobile view. For some reason you have to double click the hamburger menu to toggle it after rerouting to a page.
> 
> **Expected behaviour** Clicking on the hamburger icon toggles the menu, and it slides in from the right. When navigating to a page this goes as expected by clicking on any of the links. When you then want to navigate to a yet another route. Click on the hamburger once again, and then on the route link.
> 
> **Recreation**
> 
> 1. Load dev project
> 2. Scale to mobile view
> 3. Toggle hamburger menu
> 4. Navigate to any route
> 5. Try to navigate to another route. You should now notice you need to click the hamburger twice.

This is now fixed.

File: components/nav/Horizontal.vue [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  type Routes = Route[]
+
+  defineProps({
+    routes: { type: Array as PropType<Routes>, required: true },
+  })
+
+  defineEmits(['toggleDrawer'])
+</script>
+
+<style scoped lang="scss">
```
Comment (AasmundN) [inline components/nav/Horizontal.vue:91]: Doing custom styling like this should be avoided as much as possible. Only if you need to do small things that require media queries really.

**Review by JStoresund — APPROVED** [review]
LGTM. Eventuell videre styling kan gjøres senere. Dette ser ut som et sterkt utgangspunkt. Jeg opplever en bug dersom jeg trykker på knappene for kjapt (som du nevner i issuet), men det er detaljarbeid som kan gjøres senere. 
Skal tilstå at det er deler av koden jeg ikke forstår, men det funker knirkefritt når jeg kjører det lokalt.

File: pages/user.vue
[the remark this answers is not in the document]
Comment (AasmundN) [reply pages/user.vue:1]: Hmm, trodde jeg fikset den buggen med at du trykker på knappene for kjapt. Sikker på at du pullet de nyeste endringene?

[the remark this answers is not in the document]
Comment (AasmundN) [reply pages/user.vue:1]: > Vil denne routen eksistere hvis pages/user/index.vue finnes?

yes.

File: components/user/Button.vue
Comment (ipeglin) [inline components/user/Button.vue:1]: `user/Button` seems to be a bit too broad of a component name compared to the implementation.

File: layouts/default.vue [hunk 2]
```diff
@@ -1,13 +1,42 @@
...
-<script setup lang="ts" ?>
-  const auth = useAuthStore()
-  const isAdmin = computed(() => auth.hasAccessLevel('admin'))
+<script setup lang="ts">
+  const { mobile } = useDisplay()
+  const { currentRoute } = useRouter()
+
+  const app = useAppStore()
+
+  // public routes
```
Comment (ipeglin) [inline layouts/default.vue:20]: Did you want this to be in `nav/routes.json` og `config/routes.json`

**Review by ipeglin — COMMENTED** [review]
LGTM. Only thing I think may have to be changed in the future is the `user/Button` component

**Review by Siristol — APPROVED** [review]
Do not understand all the code but I get the general setup ( I think). It looks good when I run I it locally

File: components/user/Button.vue
Comment (AasmundN) [reply components/user/Button.vue:1]: Agreed

**Pull request merged**